website/scripts/convert_ipynb_to_mdx.py

```diff
@@ -30,526 +31,779 @@
+    return [(index, f"<CellOutput>\n{{\n  `{md}`\n}}\n</CellOutput>\n\n")]
 
-    # Begin to build the mdx string.
+
+def handle_cell_outputs(  # noqa: C901 (flake8 too complex)
+    cell: NotebookNode,
+    plot_data_folder: PathLike,
+) -> str:
+    """
+    Handle cell outputs and convert to MDX.
+
+    Args:
+        cell (NotebookNode): The cell where the outputs need converting.
+        plot_data_folder (PathLike): Path to the folder where plot data should be
+            stored.
+
+    Returns:
+        str: MDX formatted cell output.
+    """
     mdx = ""
-    # Add the frontmatter to the mdx string. This is the part between the `---` lines
-    # that define the tutorial sidebar_label information.
-    frontmatter = "\n".join(
-        ["---"]
-        + [
-            f"{key}: {value}"
-            for key, value in nb_metadata.get(
-                tutorial_folder_name,
-                {
-                    "title": "",
-                    "sidebar_label": "",
-                    "path": "",
-                    "nb_path": "",
-                    "github": "",
-                    "colab": "",
-                },
-            ).items()
-        ]
-        + ["---"]
-    )
-    frontmatter_line = len(frontmatter.splitlines())
-    mdx += f"{frontmatter}\n"
 
-    # Create the JSX and components strings.
-    jsx = ""
-    components = set()
+    # Return an empty string if there are no actual cell outputs.
+    cell_outputs = cell.get("outputs", [])
+    if not cell_outputs:
+        return mdx
+
+    # Data display priority. Below lists the priority for displaying data from cell
+    # outputs. Cells can output many different items, and some will output a fallback
+    # display, e.g. text/plain if text/html is not working. The below priorities help
+    # ensure the output in the MDX file shows the best representation of the cell
+    # output.
+    priorities = [
+        "text/markdown",
+        "image/png",  # matplotlib output.
+        "application/vnd.jupyter.widget-view+json",  # tqdm progress bars.
+        "application/vnd.bokehjs_load.v0+json",  # Bokeh loading output.
+        "application/vnd.bokehjs_exec.v0+json",  # Bokeh `show` outputs.
+        "application/vnd.plotly.v1+json",  # Plotly
+        "text/html",
+        "stream",
+        "text/plain",
+    ]
+
+    # We will use the below cell output data types for prioritizing the output shown in
+    # the MDX file.
+    cell_output_dtypes = [
+        list(cell_output["data"].keys())
+        if "data" in cell_output
+        else [cell_output["output_type"]]
+        for cell_output in cell_outputs
+    ]
+    prioritized_cell_output_dtypes = [
+        sorted(
+            set(dtypes).intersection(set(priorities)),
+            key=lambda dtype: priorities.index(dtype),
+        )
+        for dtypes in cell_output_dtypes
+    ]
+    plotly_flags = [
+        any(map(lambda output: "plotly" in output, outputs))
```

Comment:
Similar to the other place, can we rewrite this with list comprehension as well?